sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -36,29 +38,20 @@
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        List<CredentialUnavailableException> exceptions = new ArrayList<>(4);
+        StringBuilder message = new StringBuilder();
         return Flux.fromIterable(credentials)
                    .flatMap(p -> p.getToken(request).onErrorResume(CredentialUnavailableException.class, t -> {
-                       exceptions.add(t);
+                       if (!t.getClass().getSimpleName().equals("CredentialUnavailableException")) {
```

Comment:
This if statement logic is redundant now.
can be removed.

Reply:
Yeah, the onErrorResume will only get invoked if CredentialUnavailabeException is thrown.
A non CredentialUnavailabeException will never reach this block and that will automatically break the chain.
So, this check in the if statement for a CredentialUnavailabeException is not needed as a non CredentialUnavailabeException will never enter this block now.